File: src/data/services.js
import { getServiceName, getServiceSummary } from '../i18n/services'

/** Primary services — header dropdown (8 core clinical services) */
export const NAV_SERVICES = [
  { slug: 'pain-inflammation' },
  { slug: 'injuries-fractures-sprains' },
  { slug: 'arthritis' },
  { slug: 'casting-splinting' },
  { slug: 'sports-medicine' },
  { slug: 'mri-digital-imaging' },
  { slug: 'prp-orthobiologics' },
  { slug: 'car-motor-vehicle-accident-care' },
]

/** Regional / specialty services — separate pages, not in header dropdown */
export const SPECIALTY_SERVICES = [
  { slug: 'hand-wrist-care' },
  { slug: 'shoulder-elbow' },
  { slug: 'lumbar-cervical-spine' },
  { slug: 'chiropractic-surgery' },
  { slug: 'spine-surgery' },
  { slug: 'hip-knee-care' },
  { slug: 'foot-ankle-care' },
  { slug: 'total-joint-replacement' },
]

/** Full list for footer and site-wide links */
export const SERVICES = [...NAV_SERVICES, ...SPECIALTY_SERVICES]

export const WORKERS_COMP_SERVICE = {
  slug: 'workers-comp',
  href: '/workers-comp',
}

/** Every service link shown in header dropdown and footer */
export const ALL_SERVICE_LINKS = [
  { href: '/services' },
  ...SERVICES.map((service) => ({ ...service })),
  WORKERS_COMP_SERVICE,
]

/** Core services for the All Services page */
export const PRIMARY_SERVICE_CARDS = NAV_SERVICES.map((service) => ({
  ...service,
}))

/** Regional / specialty cards for the All Services page */
export const SPECIALTY_SERVICE_CARDS = SPECIALTY_SERVICES.map((service) => ({
  ...service,
}))

/** @deprecated Use PRIMARY_SERVICE_CARDS + SPECIALTY_SERVICE_CARDS */
export const SERVICE_CATALOG = [
  ...PRIMARY_SERVICE_CARDS,
  WORKERS_COMP_SERVICE,
  ...SPECIALTY_SERVICE_CARDS,
]

export function getServicePath(service) {
  if (service.href) return service.href
  return `/services/${service.slug}`
}

export function getServiceLabel(service, lang = 'en') {
  if (!service) return ''
  if (service.href === '/services') return lang === 'es' ? 'Todos los servicios' : 'All Services'
  const slug = service.slug || service.href?.replace('/services/', '')
  return getServiceName(slug, lang)
}

export function getServiceCardSummary(service, lang = 'en') {
  if (!service?.slug) return ''
  if (service.href === '/workers-comp' || service.slug === 'workers-comp') {
    return getServiceSummary('workers-comp', lang)
  }
  return getServiceSummary(service.slug, lang)
}

export { getServiceName, getServiceSummary }

File: flutter_app/lib/data/services.dart
// Service navigation — aligned with src/data/services.js

import 'service_images.dart';

class ServiceItem {
  final String slug;
  final bool isSpecialty;

  const ServiceItem({
    required this.slug,
    this.isSpecialty = false,
  });
}

const primaryServices = <ServiceItem>[
  ServiceItem(slug: 'pain-inflammation'),
  ServiceItem(slug: 'injuries-fractures-sprains'),
  ServiceItem(slug: 'arthritis'),
  ServiceItem(slug: 'casting-splinting'),
  ServiceItem(slug: 'sports-medicine'),
  ServiceItem(slug: 'mri-digital-imaging'),
  ServiceItem(slug: 'prp-orthobiologics'),
  ServiceItem(slug: 'car-motor-vehicle-accident-care'),
];

const specialtyServices = <ServiceItem>[
  ServiceItem(slug: 'hand-wrist-care', isSpecialty: true),
  ServiceItem(slug: 'shoulder-elbow', isSpecialty: true),
  ServiceItem(slug: 'lumbar-cervical-spine', isSpecialty: true),
  ServiceItem(slug: 'chiropractic-surgery', isSpecialty: true),
  ServiceItem(slug: 'spine-surgery', isSpecialty: true),
  ServiceItem(slug: 'hip-knee-care', isSpecialty: true),
  ServiceItem(slug: 'foot-ankle-care', isSpecialty: true),
  ServiceItem(slug: 'total-joint-replacement', isSpecialty: true),
];

String serviceImagePath(String slug) => ServiceImages.listImagePath(slug);

ServiceItem? getServiceBySlug(String slug) {
  for (final s in primaryServices) {
    if (s.slug == slug) return s;
  }
  for (final s in specialtyServices) {
    if (s.slug == slug) return s;
  }
  return null;
}

List<ServiceItem> get allServices => [...primaryServices, ...specialtyServices];
